docs/basic-structures.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Basic Structures
+---
+
+_Quick overview: [Basic Structures](overview.md#basic-structures)_
+
+## List
+
+Lists are an immutable, dynamically sized, homogeneous, and ordered collection
+of items. It is fast to add or remove elements from the front of a list, but
+random access is slow.
+
+Reason lists have a very basic singly-linked implementation.
+
+```reason
+let listA = [1, 2, 3];
+```
+
+Add to the front using the spread operator `...`:
+
+```reason
+let listB = [0, ...listA];
+let listC = [-1, 0, ...listA];
+```
+
+Concatenate lists with the `@` operator:
+
+```reason
+let listD = listA @ listB @ [7, 8];
+```
+
+### Common Functions
+
+- List Functions: [`module List`](https://reasonml.github.io/api/List.html)
+
+#### Map
+
+Map over a list to change items according to some function:
+
+```reason
+let doubled = List.map(i => i * 2, list);
+```
+
+#### Fold / Reduce
+
+Fold over a list to reduce it to one value (which may be another collection):
+
+```reason
+let max = List.fold_left(
+  (result, item) => item > result ? item : result,
+  initialValue,
+  list,
+);
+```
+
+#### Reverse
+
+```reason
+let reversed = List.rev(list);
+```
+
+#### Length
+
+- **Important:** This is an O(n) operation, be careful when using this function.
+It is not constant time.
+
+```reason
+let n = List.length(list);
+```
+
+#### Is Empty
+
+```reason
+let isEmpty = list == [];
+```
+
+### Pattern Matching
+
+Pattern matching works with lists:
```

Comment:
I don't think pattern matching has been introduced at this point yet. Should at least linkify the "pattern matching" word to link to the section so they can learn more.

Reply:
Yeah I'll linkify it.

I don't think there needs to be as strict a sequence of docs for the full articles, I think that will detract from the overall quality where circular references like this might be helpful (pattern matching will talk about lists, and this list doc will talk about pattern matching). In any case we can consider these topics introduced in the overview which has a non-circular sequence of topics